### PR TITLE
GitHub check runs

### PR DESCRIPTION
**Pull request by mfocko**

### CODE REVIEW

File: ogr/services/github/check_run.py [hunk 1]
```diff
@@ -0,0 +1,365 @@
...
+    """
+    Helper function for constructing valid GitHub output for check run.
+
+    Args:
+        title: Title of the output.
+        summary: Summary of the output.
+        text: Optional text for the output. Can be markdown formatted.
+
+            Defaults to `None`.
+        annotations: Optional annotations that are tied to source code.
```
Comment (lachmanfrantisek) [inline ogr/services/github/check_run.py:71]: Is the `Defaults to ..` related to `annotations or `text`?

File: ogr/services/github/check_run.py [hunk 2]
```diff
@@ -0,0 +1,365 @@
...
+    """
+    return value if value is not None else NotSet
+
+
+def create_github_check_run_output(
+    title: str,
+    summary: str,
+    text: Optional[str] = None,
+    annotations: Optional[List[Dict[str, Union[str, int]]]] = None,
+) -> GithubCheckRunOutput:
```
Comment (lachmanfrantisek) [inline ogr/services/github/check_run.py:61]: Is it really called `output` in GitHub? It really confuses me.

**Review by lachmanfrantisek — APPROVED** [review]
:tada: I don't have any objections. Nicely done! I also like all the docstrings.

LGTM once tests are passing.

File: ogr/services/github/check_run.py (hunk 1 above)
Comment (mfocko) [reply ogr/services/github/check_run.py:71]: `text` It takes double `\n\n` like GitHub to be put on separate line.

File: ogr/services/github/check_run.py (hunk 2 above)
Comment (mfocko) [reply ogr/services/github/check_run.py:61]: It is :joy: but it makes sense, you have name, some ids for internal usage, url and optionally output (title, tl;dr and markdown text)

**Comment by mfocko** [comment]
we shall pass...

**Comment by mfocko** [comment]
recheck

**Review by TomasTomecek — APPROVED** [review]
very nice! can't wait to see these in action on stg :)

**Comment by mfocko** [comment]
> very nice! can't wait to see these in action on stg :)

will require changes in service too :/

**Comment by mfocko** [comment]
recheck

File: ogr/services/github/check_run.py (hunk 1 above)
Comment (lachmanfrantisek) [reply ogr/services/github/check_run.py:71]: Aha, I need to get used to this format.

File: ogr/services/github/check_run.py (hunk 2 above)
Comment (lachmanfrantisek) [reply ogr/services/github/check_run.py:61]: Oh, let's stick with the Github naming but I don't like it TBH..;)

**Pull request merged**